### ISSUE TITLE
Background Process during image generation

### ISSUE THREAD

**Issue by mtribiere** [ShiftHackZ/Stable-Diffusion-Android#219]
**Is your feature request related to a problem? Please describe.**
During generation of an image, switching to another app or locking the phone is not possible. 

**Describe the solution you'd like**
The ideal solution in my sense would be to have the process happening in a background process. 
A permanent notification could indicate the progress of the generation process.

**Describe alternatives you've considered**
Leaving the phone on and focus on the app, but it's necessary a "convenient" solution.

**Additional context**
N/A

**Comment by PurpleCodingWizard** [comment]
Yeah, I too experience this issue. The app gets killed unless I keep it in foeground with my display turned on.

**Comment by ShiftHackZ** [comment]
Thanks for reporting this important issue. 

I think implementing generation as a background process using [Work Manager API](https://developer.android.com/topic/libraries/architecture/workmanager) that is recommended by Google for background work should resolve this.

Could you please also provide some additional information related to your user experience:
- What generation provider(s) do you use ?
- How much time does the generation process take (with the workaround you've considered) ?

**Comment by mtribiere** [comment]
I use the local generation provider, so it runs pretty slowly. I would say it takes 4~5 minutes per images for me

**Comment by ShiftHackZ** [comment]
> My phone (Asus Zenfone 10) seems to kill the app even if I run it in the foreground. I'm not sure how to disable this behavior. I have battery restrictions disabled for this app, but I'm not sure anything else I can do to prevent my phone from killing the process. I think it must have to do with the OS thinking it's using too much resources and killing it. Not sure what to do to resolve this. If anyone has any tips, I would greatly appreciate them.

Please try to disable battery optimizations for the app specifically (see the screenshot). 
Maybe it's worth checking if your manufacturer (Asus) added some additional battery optimization settings in their Android ROM, if you have any additional settings, try also disable them.

![image](https://github.com/user-attachments/assets/f5a0eb3c-09b1-4e65-b12c-b3f0bf994a98)

If it does not help, wait for the next 0.6.3 release, it will include `Background generation` feature in app settings that is based on `Android Work Manager API`, maybe this implementation will work better on your device.

**Comment by PurpleCodingWizard** [comment]
Sorry for late reply. I just saw your comment in the spam while looking for important communication in all folders.
 
> I think implementing generation as a background process using [Work Manager API](https://developer.android.com/topic/libraries/architecture/workmanager) that is recommended by Google for background work should resolve this.
Great, hopefully android doesn't kill SDAI Foss now.
 
> Could you please also provide some additional information related to your user experience:
> 
>     * What generation provider(s) do you use ?
I use the local version.

>     * How much time does the generation process take (with the workaround you've considered) ?

Depends on the image size and steps(20) but 7-10mins on average if I go 384x384 and higher. 
I usually keep the app in a floating window until SDAI gets started on the processing steps after which its usually safe (but not guranteed) to be switched to background in addition to locking the app to further reduce the possibility of being killed.

**Comment by dragos240** [comment]
> > My phone (Asus Zenfone 10) seems to kill the app even if I run it in the foreground. I'm not sure how to disable this behavior. I have battery restrictions disabled for this app, but I'm not sure anything else I can do to prevent my phone from killing the process. I think it must have to do with the OS thinking it's using too much resources and killing it. Not sure what to do to resolve this. If anyone has any tips, I would greatly appreciate them.
> 
> Please try to disable battery optimizations for the app specifically (see the screenshot). Maybe it's worth checking if your manufacturer (Asus) added some additional battery optimization settings in their Android ROM, if you have any additional settings, try also disable them.
> 
> ![image](https://private-user-images.githubusercontent.com/41302477/356604090-f5a0eb3c-09b1-4e65-b12c-b3f0bf994a98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMyMTA5NzEsIm5iZiI6MTcyMzIxMDY3MSwicGF0aCI6Ii80MTMwMjQ3Ny8zNTY2MDQwOTAtZjVhMGViM2MtMDliMS00ZTY1LWIxMmMtYjNmMGJmOTk0YTk4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA5VDEzMzc1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5ZTkyNTdhMDQ3NzdiNGYyNGMwMjRlNDJkNjZiZjZiN2ZhYjRjY2E4NGFmMjdmYmNjNGYxMDg1OGIyYTdhMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.edzVrMQVCPFIVAyIIkxqdTQvQBxD_yAZ4ca2VpIwrxo)
> 
> If it does not help, wait for the next 0.6.3 release, it will include `Background generation` feature in app settings that is based on `Android Work Manager API`, maybe this implementation will work better on your device.

I have tried everything that I was able to find on my phone and the app's setting. Everything is set to unrestricted, but my phone still kills the app. So I will wait to see if the next release fixes things

**Comment by KintCark** [comment]
U don't have enough ram the app used to do the same to me on 512x512 u can download swap file no root app from Google Play store it will help with ram issues. I just factory reset my phone and it frees up the most ram, after u factory reset it don't install to many apps the more apps you have the more ram is used.

**Comment by PurpleCodingWizard** [comment]
> U don't have enough ram the app used to do the same to me on 512x512 u can download swap file no root app from Google Play store it will help with ram issues.

You cannot create a swap file without root on android. That app is a literal scam. All it does is create a blank file in its data folder hogging storage which does nothing.
